lua/entities/terminator_nextbot_zambiemecha/shared.lua

```diff
@@ -0,0 +1,251 @@
+    
+    -- Mechanical explosion sounds
+    sound.Play( "npc/scanner/scanner_explode_crash2.wav", pos, 120, 70 )
+    sound.Play( "ambient/explosions/explode_" .. math.random( 1, 9 ) .. ".wav", pos, 120, 70 )
+    
+    -- Large explosion effect at center
+    local explode = EffectData()
+    explode:SetOrigin( pos )
+    explode:SetMagnitude( 15 )
+    explode:SetScale( 8 )
+    explode:SetRadius( radius )
+    util.Effect( "Explosion", explode )
+    
+    -- Additional helicopter bomb effects
+    for i = 1, 5 do
+        timer.Simple( i * 0.08, function()
+            local explode2 = EffectData()
+            explode2:SetOrigin( pos + VectorRand() * 80 )
+            explode2:SetMagnitude( 10 )
+            explode2:SetScale( 4 )
+            util.Effect( "HelicopterMegaBomb", explode2 )
+        end )
+    end
+    
+    -- Massive sparks and metal debris
+    for i = 1, 20 do
+        timer.Simple( math.Rand( 0, 0.5 ), function()
+            local sparkPos = pos + VectorRand() * 150
+            local sparks = EffectData()
+            sparks:SetOrigin( sparkPos )
+            sparks:SetNormal( VectorRand() )
+            sparks:SetMagnitude( 12 )
+            sparks:SetScale( 8 )
+            sparks:SetRadius( 10 )
+            util.Effect( "MetalSpark", sparks )
+        end )
+    end
+    
+    -- Fire/smoke plume
+    for i = 1, 10 do
+        timer.Simple( i * 0.1, function()
+            local smoke = EffectData()
+            smoke:SetOrigin( pos + Vector( 0, 0, i * 30 ) )
+            smoke:SetScale( 20 )
+            util.Effect( "explosion_satchel", smoke )
+        end )
+    end
+    
+    -- Damage and push entities
+    for _, ent in ipairs( ents.FindInSphere( pos, radius ) ) do
+        if not IsValid( ent ) then continue end
+        
+        local entPos = ent:GetPos()
+        local dir = ( entPos - pos ):GetNormalized()
+        local dist = entPos:Distance( pos )
+        local distFrac = 1 - ( dist / radius )
+        
+        -- Damage
+        if ent:Health() and ent:Health() > 0 then
+            local dmg = DamageInfo()
+            dmg:SetDamage( damage * distFrac )
+            dmg:SetAttacker( game.GetWorld() )
+            dmg:SetInflictor( game.GetWorld() )
+            dmg:SetDamageType( bit.bor( DMG_BLAST, DMG_BURN ) )
+            dmg:SetDamageForce( dir * 25000 * distFrac )
+            ent:TakeDamageInfo( dmg )
+        end
+        
+        -- Push
+        if ent:IsPlayer() or ent:IsNPC() or ent:IsNextBot() then
+            ent:SetVelocity( dir * 2000 * distFrac + Vector( 0, 0, 800 * distFrac ) )
+        elseif ent:GetPhysicsObject():IsValid() then
+            local phys = ent:GetPhysicsObject()
+            phys:ApplyForceCenter( dir * phys:GetMass() * 1500 * distFrac + Vector( 0, 0, phys:GetMass() * 600 * distFrac ) )
+        end
+        
+        -- Ignite nearby entities
+        if dist < radius * 0.6 and ent.Ignite then
+            ent:Ignite( 15 * distFrac )
+        end

```

Comment:
Igniting players is extremely buggy, not fun to play against. this also ignites their weapons, ensuring an instant, unfair death.
I remember being extremely conservative with the flaming zombies and igniting players, you should check them out.
Also ent.Ignite will always be valid inside an ents.FindInSphere